bigframes/dataframe.py

```diff
@@ -2719,7 +2719,8 @@ def _get_block(self) -> blocks.Block:
         return self._block
 
     def _cached(self) -> DataFrame:
-        return DataFrame(self._block.cached())
+        self._set_block(self._block.cached())
+        return self
```

Comment:
Wunderbar! This should help with taking better advantage of our cached data in the cases where we do call this automatically.